src/infrastructure/metrics.py

```
@@ -0,0 +1,234 @@
+            except Exception as e:
+                logger.error(f"Error reporting metrics: {e}")
+
+    def _report_system_metrics(self):
+        process = psutil.Process()
+        
+        self.metrics.record_gauge("memory_usage_mb", process.memory_info().rss / 1024 / 1024)
+        self.metrics.record_gauge("cpu_percent", process.cpu_percent())
+        self.metrics.record_gauge("num_threads", process.num_threads())
+        
+        disk_usage = psutil.disk_usage('/')
+        self.metrics.record_gauge("disk_usage_percent", disk_usage.percent)
```

Pylint (3.3.7)

[convention] Final newline missing

(C0304)